pkg/pillar/agentlog/loglevel.go

```diff
@@ -18,7 +17,7 @@ func GetGlobalConfig(sub *pubsub.Subscription) *types.GlobalConfig {
 		log.Infof("GlobalConfig failed %s\n", err)
 		return nil
 	}
-	gc := cast.CastGlobalConfig(m)
+	gc := m.(types.GlobalConfig)
```

Comment:
To be safe, these probably should check the type cast:

```go
gc, ok := m.(types.GlobalConfig)
if !ok {
  // do some error thing
}
```

Repeat for all of them

Reply:
We've had such checks implicitly by comparing Key() != key, but those have never fired.
And with the existing check for type in Publish*, such a cast failure would only happen when someone has an obvius bug. I prefer catching those than ignoring them with an error.
We could do if !ok {log.Fatal()} and log the type. Might make it easier for a developer.

Reply:
Yeah, that would make sense.